src/app/cart/add-product/add-product.component.ts

```diff
@@ -0,0 +1,64 @@
+import { ProductService } from '../product.service';
+import { ActivatedRoute } from '@angular/router';
+import { CartProductService } from '../../core/cart-product.service';
+import { Component, OnInit } from '@angular/core';
+import { Product } from 'src/app/models/product';
+
+@Component({
+  selector: 'app-add-product',
+  templateUrl: './add-product.component.html',
+  styleUrls: ['./add-product.component.scss'],
+})
+export class AddProductComponent implements OnInit {
+  showError: boolean;
+  showSummary: boolean;
+
+  constructor(
+    private productService: ProductService,
+    private cartService: CartProductService,
+    private route: ActivatedRoute
+  ) {
+    const productId = this.route.snapshot.params['id'];
+
+    this.process(productId);
+  }
+
+  ngOnInit(): void {}
+
+  process(productId: string) {
+    this.productService.getById(productId).subscribe(
+      (product: Product) => {
+        this.addProduct(product);
+        this.updateProduct(product);
+      },
+      () => {
+        this.showError = true;
+      }
+    );
+  }
+
+  addProduct(product: Product) {
+    this.cartService.addProduct(product).subscribe(
+      () => {
+        this.showError = false;
+        this.showSummary = true;
+      },
+      () => {
+        this.showError = true;
+      }
+    );
+  }
+
+  updateProduct(product: Product) {
+    product.stockStatus = --product.stockStatus;
```

Comment:
Normalnie to powinno być zaciągane z backendu. Dziwnie to będzie teraz wyglądać podczas testowania z dwóch przeglądarkek/użytkowników.

Reply:
albo powinno byc robione jawnie w formularzu edycji - zminiejszanie wartosci tak inkoguto troche zle pachnie :)